src/common.cc

```diff
@@ -318,15 +324,28 @@ namespace sharp {
     } else {
       if (descriptor->createChannels > 0) {
         // Create new image
-        std::vector<double> background = {
-          descriptor->createBackground[0],
-          descriptor->createBackground[1],
-          descriptor->createBackground[2]
-        };
-        if (descriptor->createChannels == 4) {
-          background.push_back(descriptor->createBackground[3]);
+        if (descriptor->createNoiseType == "gaussian") {
+          image = VImage::new_matrix(descriptor->createWidth, descriptor->createHeight);
+          std::vector<VImage> bands = {};
+          bands.reserve(descriptor->createChannels);
+          for (int _band = 0; _band < descriptor->createChannels; _band++) {
+            bands.push_back(image.gaussnoise(
+              descriptor->createWidth,
+              descriptor->createHeight,
+              VImage::option()->set("mean", descriptor->createNoiseMean)->set("sigma", descriptor->createNoiseSigma)));
+          }
+          image = image.bandjoin(bands).colourspace(VIPS_INTERPRETATION_sRGB).cast(VipsBandFormat::VIPS_FORMAT_UCHAR);
```

Comment:
I supposed that converting to (s)RGB colour space even in case for only 1 channel is an expected behaviour.